drake/doc/developers.rst

```diff
@@ -43,6 +43,42 @@ available in the superbuild.
 class method runs as a unit test, and fails if it times out or calls
 ``error``. For more information, see ``drake/util/unitTest.m``.
 
+Supported Configurations
+========================
+
+The following table shows the configurations and platforms that Drake
+officially supports. Supported configurations are tested in continuous
+integration. All other configurations are provided on a best-effort basis.
+On Ubuntu and OS X, only makefile builds are supported.
```

Comment:
The term "generator" in the context of CMake refers to which type of native build system CMake emits (CMake is a meta-build system -- a program for describing how to produce other build systems).  The line above covers your question -- on Ubuntu and OS X, only the "Makefile" generator is supported.  On Windows, there is a VSProj generator (or whatever its called) that I suspect is in the supported set, possibly in addition to Makefiles.